phpunit/class-wp-rest-navigation-fallbacks-controller-test.php

```diff
@@ -0,0 +1,186 @@
+
+		$this->assertEquals( 'rest_cannot_create', $data['code'], 'Response should indicate user cannot create.' );
+
+		$this->assertEquals( 'Sorry, you are not allowed to create Navigation Menus as this user.', $data['message'], 'Response should indicate failed request status.' );
+	}
+
+	/**
+	 * @covers WP_REST_Navigation_Fallbacks_Controller
+	 *
+	 * @since 6.3.0 Added Navigation Fallbacks endpoint.
+	 */
+	public function test_should_return_fallback_navigation_menu() {
+
+		$request  = new WP_REST_Request( 'GET', '/wp-block-editor/v1/navigation-fallbacks' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( 200, $response->get_status(), 'Status should indicate successful request.' );
+
+		$this->assertIsArray( $data );
+
+		$this->assertArrayHasKey( 'id', $data );
+
+		$this->assertEquals( 'wp_navigation', get_post_type( $data['id'] ) );
+
+		// Check that only a single Navigation fallback was created.
+		$navs_in_db = $this->get_navigations_in_database();
+
+		$this->assertCount( 1, $navs_in_db, 'Only the existing Navigation menus should be present in the database.' );
+
+	}
+
+	/**
+	 * @covers WP_REST_Navigation_Fallbacks_Controller
+	 *
+	 * @since 6.3.0 Added Navigation Fallbacks endpoint.
+	 */
+	public function test_should_provide_schema() {
+		$request  = new WP_REST_Request( 'OPTIONS', '/wp-block-editor/v1/navigation-fallbacks' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( 200, $response->get_status(), 'Status should indicate successful request.' );
+
+		$this->assertArrayHasKey( 'schema', $data, '"schema" key should exist in response.' );
+
+		$schema = $data['schema'];
+
+		$this->assertEquals( 'object', $schema['type'] );
+
+		$this->assertArrayHasKey( 'id', $schema['properties'], 'Schema should have an "id" property.' );
+		$this->assertEquals( 'integer', $schema['properties']['id']['type'], 'Schema "id" property should be an integer.' );
+		$this->assertTrue( $schema['properties']['id']['readonly'], 'Schema "id" property should be readonly.' );
+	}
+
+	private function get_navigations_in_database() {
+		$navs_in_db = new WP_Query(
+			array(
+				'post_type'      => 'wp_navigation',
+				'post_status'    => 'publish',
+				'posts_per_page' => -1,
+				'orderby'        => 'date',
+				'order'          => 'DESC',
+			)
+		);
+
+		return $navs_in_db->posts ? $navs_in_db->posts : array();
+	}
+
+	/**
+	 * @doesNotPerformAssertions
+	 */
+	public function test_prepare_item() {
+		// Covered by the core test.
+	}
+
+	/**
+	 * @doesNotPerformAssertions
+	 */
+	public function test_context_param() {
+		// Covered by the core test.
+	}
+
+	/**
+	 * @doesNotPerformAssertions
+	 */
+	public function test_get_items() {
+		// Covered by the core test.
+	}
+
+	/**
+	 * @doesNotPerformAssertions
+	 */
+	public function test_get_item() {
+		// Controller does not implement get_item().
```

Comment:
Controller does implement get_item.

Reply:
Ah yes. We renamed the method. I'll swap it out.

Reply:
✅